hazelcast/src/main/java/com/hazelcast/map/impl/operation/steps/engine/StepSupplier.java

```diff
@@ -200,7 +200,12 @@ private void rerunWithForcedEviction(Runnable step) {
 
     public void handleOperationError(Throwable throwable) {
         state.setThrowable(throwable);
-        UtilSteps.HANDLE_ERROR.runStep(state);
+        currentRunnable = null;
+        currentStep = UtilSteps.HANDLE_ERROR;
```

Comment:
Why TxnSomethingOpSteps sends response in the runStep but StepSupplier.handleOperationError() changed to be like this?

Reply:
There was no offloading on step-chain of TxnSomethingOpSteps and for this reason, creating a new step has no purposem, all steps can be merged into one. So i did inline it. 

Regarding `handleOperationError` change, picking next step must be like that to align it with next-step finding.execution logic. Other step execution logic can be seen for verification.

Reply:
> There was no offloading on step-chain of TxnSomethingOpSteps and for this reason, creating a new step has no purposem, all steps can be merged into one. So i did inline it.

I understand this now.

> Regarding handleOperationError change, picking next step must be like that to align it with next-step finding.execution logic. Other step execution logic can be seen for verification.

I also understand that the change makes `HANDLE_ERROR` as next step.

But I don't understand why we can't continue using `UtilSteps.HANDLE_ERROR.runStep(state);`.

Is it because previous step that throw an exception was offloaded and we want HANDLE_ERROR step to run on partition threads? Or is there an another reason?

Reply:
To dispose resource on a failure, like HD memory free,  we want to run HANDLE_ERROR on partition threads.